cpp/bridges_module/bridges_test.cpp

```diff
@@ -0,0 +1,213 @@
+#include <queue>
+
+#include <gtest/gtest.h>
+#include <mg_generate.hpp>
+#include <mg_graph.hpp>
+#include <mg_test_utils.hpp>
+
+#include "algorithm/bridges.hpp"
+
+namespace {
+
+std::uint64_t CountComponents(const mg_graph::Graph<> &graph) {
+  std::uint64_t n_components = 0;
+  std::vector<bool> visited(graph.Nodes().size());
+
+  for (auto &node : graph.Nodes()) {
+    if (visited[node.id]) continue;
+    ++n_components;
+
+    std::queue<std::uint64_t> queue;
+    visited[node.id] = true;
+    queue.push(node.id);
+    while (!queue.empty()) {
+      auto curr_id = queue.front();
+      queue.pop();
+      for (const auto &neigh : graph.Neighbours(curr_id)) {
+        auto next_id = neigh.node_id;
+        if (visited[next_id]) continue;
+        visited[next_id] = true;
+        queue.push(next_id);
+      }
+    }
+  }
+  return n_components;
+}
+
+std::vector<std::pair<std::uint64_t, std::uint64_t>> GetBridgesBruteforce(mg_graph::Graph<> *graph) {
+  using IntPair = std::pair<std::uint64_t, std::uint64_t>;
+
+  std::uint64_t comp_cnt = CountComponents(*graph);
+  std::vector<IntPair> bridges;
+  auto edges = graph->ExistingEdges();
+  for (const auto &edge : edges) {
+    auto from = edge.from, to = edge.to;
+
+    graph->EraseEdge(from, to);
+    auto new_comp_cnt = CountComponents(*graph);
+
+    if (new_comp_cnt > comp_cnt) bridges.emplace_back(from, to);
+    graph->CreateEdge(from, to);
+  }
+  return bridges;
+}
+
+/// Checks if obtained list of bridges is correct.
+bool CheckBridges(std::vector<mg_graph::Edge<>> user, std::vector<std::pair<std::uint64_t, std::uint64_t>> correct) {
+  std::set<std::pair<std::uint64_t, std::uint64_t>> user_bridge_set, correct_bridge_set;
+  for (auto &p : correct) {
+    correct_bridge_set.insert(p);
+    correct_bridge_set.insert({p.second, p.first});
+  }
+  for (auto &edge : user) {
+    user_bridge_set.insert({edge.from, edge.to});
+    user_bridge_set.insert({edge.to, edge.from});
```

Comment:
```suggestion
    user_bridge_set.emplace_back(edge.from, edge.to);
    user_bridge_set.emplace_back(edge.to, edge.from);
```

Reply:
Are you sure that std::set contains this function (_emplace_back_)? Also, we are working with pairs here

Reply:
You are right its `emplace`, sry :smile: